src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -1,715 +1,29 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
-
-using System;
-using System.Collections.Concurrent;
+﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
-using System.Threading;
-using System.Threading.Tasks;
+using System.Web;
 using System.Web.Configuration;
-using NuGet.Server.Logging;
 
 namespace NuGet.Server.Infrastructure
 {
-    /// <summary>
-    /// ServerPackageRepository represents a folder of nupkgs on disk. All packages are cached during the first request in order
-    /// to correctly determine attributes such as IsAbsoluteLatestVersion. Adding, removing, or making changes to packages on disk 
-    /// will clear the cache.
-    /// </summary>
-    public class ServerPackageRepository
-        : PackageRepositoryBase, IServerPackageRepository, IPackageLookup, IDisposable
+	public class ServerPackageRepository : ServerPackageRepositoryCore
```

Comment:
That's a great insight. I agree that your alternative is correct, and I'd like to take a day to think about it.

Here are some preliminary thoughts, feel free to chime in. CC @harikm86 @xavierdecoster 

We do know people are using the code as basis for enhancements (like you just did) so it is somewhat of a concern, at the same time folks that make changes should be able to address them if we release note them.